src/components/dialog/GitLogDialog.vue

```diff
@@ -0,0 +1,149 @@
+                <q-item-section>
+                  {{ truncateCommitMessage(item.commit.message) }}
+                </q-item-section>
+              </template>
+              <q-list dense class="bg-black text-white">
+                <q-item>
+                  <q-item-section avatar>
+                    <q-avatar icon="fa-solid fa-hashtag"/>
+                  </q-item-section>
+                  <q-item-section>
+                    {{item.oid}}
+                  </q-item-section>
+                </q-item>
+                <q-item>
+                  <q-item-section avatar>
+                    <q-avatar icon="fa-solid fa-user"/>
+                  </q-item-section>
+                  <q-item-section>
+                    {{item.commit.author.name}} &#60;{{item.commit.author.email}}&#62;
+                  </q-item-section>
+                </q-item>
+                <q-item>
+                  <q-item-section avatar>
+                    <q-avatar icon="fa-solid fa-calendar"/>
+                  </q-item-section>
+                  <q-item-section>
+                    {{new Date(item.commit.committer.timestamp * 1000).toLocaleString()}}
+                  </q-item-section>
+                </q-item>
+                <q-item>
+                  <q-item-section avatar>
+                    <q-avatar icon="fa-solid fa-message"/>
+                  </q-item-section>
+                  <q-item-section>
+                    <p v-html="$sanitize(item.commit.message.replaceAll('\n', '<br>'))"></p>
+                  </q-item-section>
+                </q-item>
+              </q-list>
+            </q-expansion-item>
+          </template>
+        </q-list>
+        <template v-slot:loading>
+          <div class="row justify-center q-my-md">
+            <q-spinner-dots color="primary" size="40px" />
+          </div>
+        </template>
+      </q-infinite-scroll>
+      </div>
+    </template>
+  </default-dialog>
+</template>
+
+<script setup>
+import { onMounted, onUnmounted, ref } from 'vue';
+import DefaultDialog from 'components/dialog/DefaultDialog';
+import DialogEvent from 'src/composables/events/DialogEvent';
+import { gitLog } from 'src/composables/Project';
+
+const props = defineProps({
+  projectName: {
+    type: String,
+    required: true,
+  },
+});
+const scrollTargetRef = ref(null);
+const logItems = ref([]);
+const branchName = ref(null);
+let dialogEventSubscription;
+
+/**
+ * Load logs.
+ * @param {Number} index - The index parameter can be used to make some sort of pagination on the
+ * content you load. It takes numeric values starting with 1 and incrementing with each call
+ * @param {Function} done - Function to call when you made all necessary updates. DO NOT forget
+ * to call it otherwise your loading message will continue to be displayed.
+ * @see https://quasar.dev/vue-components/infinite-scroll
+ */
+function onLoad(index, done) {
+  let commitRef = branchName.value;
+  let depth = 25;
```

Comment:
why 25 here and 10 as default value for `depth` ?

Reply:
I change 10 by 25 in Project